designs/tinkerbell-provider-input.md

```diff
@@ -0,0 +1,366 @@
+
+This new command will take a CSV file as input with configurations for all the hardware, parse it and output the required hardware manifests. This will also by default push the hardware to the Tinkerbell stack.
+
+The CSV file will need these headers to identify which field corresponds to which value:
+* ip_address: IP address to configure on the machine
+* gateway: Gateway to configure on the machine
+* nameservers: List of nameservers to be specified on the machine, delimited by `|`
+* netmask: Netmask for the IP
+* mac: MAC address of the interface to assign the IP to
+* hostname: Hostname of the machine
+* vendor: Name of the machine's vendor
+* bmc_ip: IP address of the machine's BMC interface
+* bmc_username: Username for the machine's BMC
+* bmc_password: Password for machine's BMC
+
+All of these fields are required and can be placed in any order in the CSV file.
+
+Here is an example hardware CSV file with the required information for two hardwares:
+```csv
+ip_address,gateway,nameservers,netmask,mac,hostname,vendor,bmc_ip,bmc_username,bmc_password
+192.168.1.10,192.168.0.1,8.8.8.8|8.8.4.4,255.255.255.0,00:00:00:00:00:00,worker1,supermicro,192.168.0.10,admin,Admin
+192.168.1.11,192.168.0.1,8.8.8.8|8.8.4.4,255.255.255.0,00:00:00:00:00:01,worker2,supermicro,192.168.0.11,admin,Admin
+```
+
+Once the CSV file is set up, users can use the `setup hardware` command to generate and push the hardware manifests
+
+(Recommended) Command to generate the hardware manifests and push it to the Tinkerbell stack
+```bash
+eksctl anywhere setup hardware -f <path-to-csv-file> --tinkerbell-ip <tinkerbell-stack-ip>
+```
+
+(NOT Recommended) Command to just generate the hardware manifests without pushing
+```bash
+eksctl anywhere generate hardware -f <path-to-csv-file> --dry-run
+```
+
+Running the above commands is going a create a new directory called `hardware-manifests` with the manifests. This is what the directory structure will look like:
+```
+hardware-manifests
+├── hardware.yaml
+└── json
+    ├── worker1.json
+    └── worker2.json
+```
+
+### Using the generated files to create a cluster
+
+Once the users have created the necessary files, they can use them to create an EKS Anywhere Tinkerbell cluster.
+
+```bash
+eksctl anywhere create cluster -f <clusterconfig> -w <hardware YAML>
+```
+
+## Future changes
+
+Most of the design details described above are only valid for beta release of Tinkerbell provider as for beta, we are using the default installation of Tinkerbell. This default installation runs Tinkerbell components as microservices on docker containers. We have plans to eliminate Tinkerbell's dependency on docker and instead run it on Kubernetes as controllers. You can read the proposal for this [here](https://github.com/tinkerbell/proposals/tree/main/proposals/0026).
+
+Since this is a work-in-progress, we still have not finalized how our design will look like when we leverage Kubernetes backend for Tinkerbell.
```

Comment:
Can we talk about what we discussed in a meeting earlier about how these commands are going to change? For example changes that can be made to upstream to decrease the number of files we need to maintain and changing our abstraction of these objects?

Reply:
Added a few more specific things that we expect will change